google-cloud-spanner/lib/google/cloud/spanner/session.rb

```diff
@@ -88,6 +88,26 @@ def path
           @grpc.name
         end
 
+        ##
+        # Cloud Labels are a flexible and lightweight mechanism for organizing
+        # cloud resources into groups that reflect a customer's organizational
+        # needs and deployment strategies. Cloud Labels can be used to filter
+        # collections of resources. They can be used to control how resource
+        # metrics are aggregated. And they can be used as arguments to policy
+        # management rules (e.g. route, firewall, load balancing, etc.).
+        #
+        # * Label keys must be between 1 and 63 characters long and must conform
+        #   to the following regular expression: `[a-z]([-a-z0-9]*[a-z0-9])?`.
+        # * Label values must be between 0 and 63 characters long and must
+        #   conform to the regular expression `([a-z]([-a-z0-9]*[a-z0-9])?)?`.
+        # * No more than 64 labels can be associated with a given resource.
+        #
+        # @return [Hash{String=>String}] The label keys and values in a hash.
+        #
+        def labels
+          @grpc.labels
```

Comment:
This will return a `Google::Protobuf::Map` object. It is probably better to call `@grpc.labels.to_h` instead.

Reply:
Good catch. Does this need to change in `Instance#labels` too?

Reply:
Yes, especially if those labels are being passed into a Protobuf constructor, because the constructor won't accept a Map, it only takes a Hash.

Reply:
Although if we do convert the Instance Map to a Hash, we may need to provide lifecycle management for the Hash, since the Hash could be updated and `Instance#save` called.

Reply:
Another option is to keep the values using Map, but always cast Map to Hash when passing to the API.

```ruby
if labels
  labels_hash = Hash[labels.map { |key, value| [String(key), String(value)] }]
  session = Google::Spanner::V1::Session.new labels: labels_hash 
end
```